include/swift/AST/Decl.h

```diff
@@ -7941,7 +7941,12 @@ inline bool Decl::isPotentiallyOverridable() const {
       isa<FuncDecl>(this) ||
       isa<DestructorDecl>(this)) {
     auto classDecl = getDeclContext()->getSelfClassDecl();
-    return classDecl && !classDecl->isActor();
+    if (auto *VD = dyn_cast<ValueDecl>(this)) {
```

Comment:
You can make this an unconditional cast since all of the types checked for above are ValueDecl.